### ISSUE TITLE
New SDL causes Xbox trigger complication

### ISSUE THREAD

**Issue by koga-ninja** [libsdl-org/SDL#7940]
When updating the sdl file for AntimicroX, commands assigned to the left and right triggers do not work unless another input is held down at the same time.

**Comment by slouken** [comment]
Does it work if you press the A button first and then press the triggers?

**Comment by koga-ninja** [comment]
Yes, exactly. But it works if I hold down B, Y, etc. too.

**Comment by slouken** [comment]
Okay, this is expected. SDL is using raw input which allows it to see the name of controllers and support more than 4 controllers, but has the downside that it can't do rumble until it sees a button press and knows which XInput slot is associated with the controller.

**Comment by slouken** [comment]
Actually, what version of SDL are you using with AntimicroX?

**Comment by slouken** [comment]
I'm testing with AntimicroX 3.3.4, which uses SDL 2.26.5, and the triggers show up fine in the controller test UI. Is that what you're seeing?

**Comment by slouken** [comment]
I'm able to reproduce this, I'm investigating.

**Comment by slouken** [comment]
It's because of b2e88ecfeb, which was a workaround for Xbox controllers disconnecting when using the wireless adapter.

**Comment by koga-ninja** [comment]
Yes I figured it may be related. It's basically a choice between the Xbox controller being able to reconnect or having the triggers function properly at this point? Thank you for looking into this.

**Comment by slouken** [comment]
I found a workaround!

**Comment by koga-ninja** [comment]
> I found a workaround!

Thank you!!!

**Comment by kanedauy** [comment]
@slouken Yesss, I need this fix in AntimicroX!

**Comment by koga-ninja** [comment]
> I found a workaround!

@slouken Will this be included in the next release? 2.28.2?

**Comment by slouken** [comment]
Yep!

**Comment by slouken** [comment]
2.28.2 has been released. :)

**Comment by slouken** [comment]
You're welcome! :)
